File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/CurrencyMapper.java
package org.mifos.mobilewallet.core.data.fineract.entity.mapper;

import org.mifos.mobilewallet.core.domain.model.Currency;

import javax.inject.Inject;

/**
 * Created by naman on 17/8/17.
 */

public class CurrencyMapper {

    @Inject
    CurrencyMapper() {
    }

    public Currency transform(org.mifos.mobilewallet.core.data.fineract
            .entity.accounts.savings.Currency savingsCurrency) {

        Currency currency = new Currency();
        currency.setCode(savingsCurrency.getCode());
        currency.setDisplayLabel(savingsCurrency.getDisplayLabel());
        currency.setDisplaySymbol(savingsCurrency.getDisplaySymbol());

        return currency;
    }
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/CurrencyMapper.kt
package org.mifos.mobilewallet.core.data.fineract.entity.mapper

import org.mifos.mobilewallet.core.data.fineract.entity.accounts.savings.Currency
import javax.inject.Inject




/**
 * Created by naman on 17/8/17.
 */
class CurrencyMapper @Inject constructor(){

    fun transform(savingsCurrency: Currency): org.mifos.mobilewallet.core.domain.model.Currency {
        val currency = org.mifos.mobilewallet.core.domain.model.Currency()
        currency.code = savingsCurrency.code
        currency.displayLabel = savingsCurrency.displayLabel
        currency.displaySymbol = savingsCurrency.displaySymbol
        return currency
    }
}